File: exercises/r/01_exploracion_superstore.R
# =====================================================
# Ejercicio 1
# Exploración dataset Superstore
# Lenguaje: R
# =====================================================


# Librerías

library(readr)
library(dplyr)


# -----------------------------------------------------
# 1. Cargar dataset
# -----------------------------------------------------

ventas <- read_csv(
  "datasets/superstore.csv"
)


# -----------------------------------------------------
# 2. Primeros registros
# -----------------------------------------------------

cat("\nPrimeros registros\n")

print(
  head(ventas)
)


# -----------------------------------------------------
# 3. Dimensiones del dataset
# -----------------------------------------------------

cat("\nDimensiones del dataset\n")

print(
  dim(ventas)
)


# -----------------------------------------------------
# 4. Nombre de columnas
# -----------------------------------------------------

cat("\nColumnas disponibles\n")

print(
  names(ventas)
)


# -----------------------------------------------------
# 5. Estructura del dataset
# -----------------------------------------------------

cat("\nEstructura del dataset\n")

str(ventas)


# -----------------------------------------------------
# 6. Resumen estadístico
# -----------------------------------------------------

cat("\nResumen estadístico\n")

summary(ventas)


# -----------------------------------------------------
# 7. Categorías disponibles
# -----------------------------------------------------

cat("\nCategorías\n")

print(
  unique(ventas$Category)
)


# -----------------------------------------------------
# 8. Regiones disponibles
# -----------------------------------------------------

cat("\nRegiones\n")

print(
  unique(ventas$Region)
)

File: exercises/python/01_exploracion_superstore.py
# =====================================================
# Ejercicio 1
# Exploración dataset Superstore
# =====================================================

import pandas as pd


# Cargar datos

ventas = pd.read_csv(
    "datasets/superstore.csv",
    encoding="latin-1"
)

# Primeras filas

print("\nPrimeros registros")
print(ventas.head())


# Dimensiones

print("\nCantidad de filas y columnas")
print(ventas.shape)


# Columnas

print("\nColumnas del dataset")
print(ventas.columns)


# Información general

print("\nInformación del dataset")
ventas.info()


# Estadísticas numéricas

print("\nEstadísticas")
print(ventas.describe())


# Categorías existentes

print("\nCategorías")
print(ventas["Category"].unique())


# Regiones existentes

print("\nRegiones")
print(ventas["Region"].unique())